src/components/Trade/TradeConfirm/ReceiveSummary.tsx

```diff
@@ -7,35 +7,47 @@ import {
   useColorModeValue,
   useDisclosure,
 } from '@chakra-ui/react'
-import { type FC } from 'react'
+import type { ChainAdapterManager } from '@shapeshiftoss/chain-adapters'
+import { type FC, useMemo } from 'react'
 import { useTranslate } from 'react-polyglot'
 import { Amount } from 'components/Amount/Amount'
 import { HelperTooltip } from 'components/HelperTooltip/HelperTooltip'
 import { type RowProps, Row } from 'components/Row/Row'
 import { RawText, Text } from 'components/Text'
+import { getChainAdapterManager } from 'context/PluginProvider/chainAdapterSingleton'
 import { bnOrZero } from 'lib/bignumber/bignumber'
+import { fromBaseUnit } from 'lib/math'
+import type { AmountDisplayMeta } from 'lib/swapper/api'
 import { SwapperName } from 'lib/swapper/api'
 
 type ReceiveSummaryProps = {
   isLoading?: boolean
   symbol: string
-  amount: string
-  intermediaryTransactionOutputs?: { amount: string; symbol: string; chainName?: string }[]
+  amountCryptoPrecision: string
+  intermediaryTransactionOutputs?: AmountDisplayMeta[]
   fiatAmount?: string
-  beforeFees?: string
-  protocolFee?: string
+  amountBeforeFeesCryptoPrecision?: string
+  protocolFees?: AmountDisplayMeta[]
   shapeShiftFee?: string
   slippage: string
   swapperName: string
 } & RowProps
 
+const parseAmountDisplayMeta =
+  (chainAdapterManager: ChainAdapterManager) =>
```

Comment:
Any reason why we inject this? This is a singleton, thus should have no perf. implications to get from within this method